scripts/ban-nondeterminism.sh

```
@@ -0,0 +1,110 @@
+  # '\bstd::sync::RwLock\b'
+  # '\bstd::sync::atomic::\b'
+)
+
+echo "ban-nondeterminism: scanning paths:"
+for p in $PATHS; do echo "  - $p"; done
+echo
+
+violations=0
+for pat in "${PATTERNS[@]}"; do
+  echo "Checking: $pat"
+  if rg "${RG_ARGS[@]}" "${ALLOW_GLOBS[@]}" -n -S "$pat" $PATHS; then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)